src/main/java/frc/trigon/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,183 @@
+package frc.trigon.robot.subsystems.arm;
+
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+import com.ctre.phoenix6.hardware.CANcoder;
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.*;
+import frc.trigon.robot.utilities.Conversions;
+
+public class Arm extends SubsystemBase {
+    private final static Arm INSTANCE = new Arm();
+
+    private final CANSparkMax
+            masterAngleMotor = ArmConstants.MASTER_ANGLE_MOTOR,
+            followerAngleMotor = ArmConstants.FOLLOWER_ANGLE_MOTOR,
+            masterElevatorMotor = ArmConstants.MASTER_ELEVATOR_MOTOR,
+            followerElevatorMotor = ArmConstants.FOLLOWER_ELEVATOR_MOTOR;
+    private final CANcoder angleEncoder = ArmConstants.ANGLE_ENCODER;
+    private final TalonSRX elevatorEncoder = ArmConstants.ELEVATOR_ENCODER;
+    private TrapezoidProfile
+            angleMotorProfile = null,
+            elevatorMotorProfile = null;
+    private double
+            lastAngleMotorProfileGenerationTime,
+            lastElevatorMotorProfileGenerationTime;
+
+    public static Arm getInstance() {
+        return INSTANCE;
+    }
+
+    private Arm() {
+    }
+
+    public Command getSetArmTarget(ArmConstants.ArmState targetState){
+        if (getElevatorPositionRevolutions() < targetState.elevatorPosition) {
```

Comment:
There is a slight problem here. The command getter will be called at the start of the code, not when the command should be run. This means `getElevatorPositionRevolutions() < targetState.elevatorPosition` won't be dynamic to the current position, just a statement decided at the start of the code. To solve this, use a DefferedCommand. Read a bit about it.

Reply:
couldn't I just make it a supplier?

Reply:
I can't find anything about DefferedCommand

Reply:
1. Due to command usage this can't be a supplier, but the idea of it is that it functions like one. 
2. Type DeferredCommand (sorry had a type) in intellij and read its javadocs

Reply:
I can't find anything on DeferredCommand  in intelij. I typed it in and it didn't recognize it. I tried searching through all the files and still nothing.

Reply:
Are you sure you updated from main? I type to do it in the group.

Reply:
yeah

Reply:
Try to type DeferredCommand and alt + enter to import it.

Reply:
![image](https://github.com/Programming-TRIGON/RobotCodePreSeason2024/assets/142012009/a3e75465-63f6-4676-a6de-70d80f0077e8)
tried.

Reply:
Did you fetch origin before updating from main?

Reply:
yeah

Reply:
Send me your build.gradle file in discord.

Reply:
do I use a DeferedCommnd instead of the SequentialCommandGroup? if not, where do I use it?

Reply:
also, why does this not work? I read about the class and it said it needs a supplier of a command and requierments. so what;s wrong?
![image](https://github.com/Programming-TRIGON/RobotCodePreSeason2024/assets/142012009/fbd43f75-8ac1-47b7-a01d-3f61c55f7409)